src/Compilers/Shared/CoreClrShim.cs

```diff
@@ -53,5 +53,15 @@ internal static void RegisterProvider(object provider)
                 }
             }
         }
+
+        internal static class FileNotFoundException
```

Comment:
This particular file is meant for reflection specific to CoreClr.  This usage is the opposite, reflection specific to the desktop.  I think we should move this functionality to a file named DesktopShim.